### PR TITLE
ceph.spec.in: move remaining specific BuildRequires out of common

### PR DESCRIPTION
**Pull request by smithfarm**
Move both remaining distro-specific BuildRequires from the "common" section
to the "specific" section.

### CODE REVIEW

File: ceph.spec.in [hunk 1]
```diff
@@ -81,24 +81,23 @@ BuildRequires:	python
...
 BuildRequires:	yasm
-%if 0%{?rhel} || 0%{?fedora} || 0%{?suse_version}
-BuildRequires:	snappy-devel
-%endif
 
 #################################################################################
 # specific
 #################################################################################
-%if ! 0%{?rhel} || 0%{?fedora}
+%if ! 0%{?rhel} || 0%{?fedora} || 0%{?suse_version}
```
Comment (dachary) [inline ceph.spec.in]: snappy-devel is also for rhel (see above)
<pre>
%if 0%{?rhel} || 0%{?fedora} || 0%{?suse_version}
</pre>

Comment (smithfarm) [inline ceph.spec.in]: @ktdreyer assuming the `!` operator has higher precedence than `||`, the condition `%if ! 0%{?rhel} || 0%{?fedora}` evaluates to true for *any* distro for which `%{rhel}` is false. Correct?

File: ceph.spec.in [hunk 2]
```diff
@@ -81,24 +81,23 @@ BuildRequires:	python
...
 BuildRequires:	sharutils
+BuildRequires:	snappy-devel
+%endif
+
+%if ( 0%{?rhel} > 0 && 0%{?rhel} < 7 ) || ( 0%{?centos} > 0 && 0%{?centos} < 7 )
```
Comment (ktdreyer) [inline ceph.spec.in]: No need to check `%{centos}` here; the CentOS project's redhat-rpm-macros package will set the `%{rhel}` macro the same way that RHEL itself does.

File: ceph.spec.in (hunk 1 above)
Comment (ktdreyer) [inline ceph.spec.in]: Yeah, you can just remove the `!` operator and use the line that @dachary wrote above. RHEL ships `sharutils` (and `snappy-devel`), so there's no need to specify "if-not-rhel" here.

**Comment by smithfarm** [comment]
@ktdreyer rebased to current master and made the changes you and @dachary suggested

**Comment by ktdreyer** [comment]
pushed to `wip-master-specfile-cleanup3` to verify on gitbuilders

**Comment by smithfarm** [comment]
@ktdreyer: sorry, could you take another look? It occurred to me that a conditional that says "if rhel/centos or fedora or suse" doesn't really make sense in a spec file - or are we supporting any other RPM-based distros besides these?

File: ceph.spec.in [hunk 3]
```diff
@@ -81,24 +81,20 @@ BuildRequires:	python
 BuildRequires:	python-argparse
 BuildRequires:	python-nose
 BuildRequires:	python-requests
-%if ( 0%{?rhel} > 0 && 0%{?rhel} < 7 ) || ( 0%{?centos} > 0 && 0%{?centos} < 7 )
-BuildRequires:	python-sphinx10
-%endif
 BuildRequires:	python-virtualenv
+BuildRequires:	sharutils
```
Comment (osynge) [inline ceph.spec.in]: this causes a build fail on fedora

File: ceph.spec.in [hunk 4]
```diff
@@ -81,24 +81,20 @@ BuildRequires:	python
...
-%if 0%{?rhel} || 0%{?fedora} || 0%{?suse_version}
-BuildRequires:	snappy-devel
-%endif
 
 #################################################################################
 # specific
 #################################################################################
-%if ! 0%{?rhel} || 0%{?fedora}
-BuildRequires:	sharutils
+%if 0%{?rhel} > 0 && 0%{?rhel} < 7
```
Comment (osynge) [inline ceph.spec.in]: I think this is the wrong macros for family_redhat or family_suse

File: ceph.spec.in [hunk 5]
```diff
@@ -81,24 +81,20 @@ BuildRequires:	python
 BuildRequires:	python-argparse
 BuildRequires:	python-nose
 BuildRequires:	python-requests
-%if ( 0%{?rhel} > 0 && 0%{?rhel} < 7 ) || ( 0%{?centos} > 0 && 0%{?centos} < 7 )
-BuildRequires:	python-sphinx10
-%endif
```
Comment (osynge) [inline ceph.spec.in]: These macros for family_redhat or family_suse seem more specific, is their a reason for this change.

Comment (smithfarm) [inline ceph.spec.in]: Yes, @ktdreyer told me to make that change in https://github.com/ceph/ceph/pull/4881#discussion-diff-31933581 -- github is hiding this under the cryptic message "ktdreyer commented on an outdated diff"

File: ceph.spec.in (hunk 3 above)
Comment (smithfarm) [inline ceph.spec.in]: ok I will take a look

**Comment by smithfarm** [comment]
closing in favor of  #4897, #4898, and #4906

**Pull request closed without merging**